### PR TITLE
Add ability to configure remaining defaults in PeerManager

### PR DESCRIPTION
**Pull request by agunde406**
Adds options to the PeerManagerBuilder to configure retry_frequency,
max_retry_frequency, and endpoint_retry_frequency. These values
were previously only set to the default values.

Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

### CODE REVIEW

File: libsplinter/src/peer/builder.rs [hunk 1]
```diff
@@ -75,6 +84,33 @@ impl PeerManagerBuilder {
...
+    /// How often (in seconds) the `PeerManager` will  wait before retrying a pending
+    /// peers endpoints
+    pub fn with_retry_frequency(mut self, retry_frequency: u64) -> Self {
+        self.retry_frequency = Some(retry_frequency);
+        self
+    }
+
+    /// Set the max_retry_frequency to use with the resulting `PeerManager`.
+    ///
+    /// The maximum (in seconds) the `PeerManager` will wait before retrying a pending
```
Comment (ltseeley) [inline libsplinter/src/peer/builder.rs:98]: s/maximum/maximum time/

File: libsplinter/src/peer/builder.rs [hunk 2]
```diff
@@ -75,6 +84,33 @@ impl PeerManagerBuilder {
...
+    /// peers endpoints
+    pub fn with_max_retry_frequency(mut self, retry_frequency: u64) -> Self {
+        self.max_retry_frequency = Some(retry_frequency);
+        self
+    }
+
+    /// Set the initial endpoint_retry_frequency to use with the resulting `PeerManager`.
+    ///
+    /// How often (in seconds) the `PeerManager` will wait before retrying a pending
+    /// requested endpoints that does not have and id
```
Comment (ltseeley) [inline libsplinter/src/peer/builder.rs:108]: s/endpoints/endpoint/ and s/and id/an ID/

Comment (agunde406) [reply libsplinter/src/peer/builder.rs]: fixed

Comment (ltseeley) [reply libsplinter/src/peer/builder.rs]: Still needs s/and/an/

Comment (agunde406) [reply libsplinter/src/peer/builder.rs]: fixed

File: libsplinter/src/peer/builder.rs [hunk 3]
```diff
@@ -75,6 +84,33 @@ impl PeerManagerBuilder {
         self
     }
 
+    /// Set the initial retry_frequency to use with the resulting `PeerManager`.
+    ///
+    /// How often (in seconds) the `PeerManager` will  wait before retrying a pending
```
Comment (jsmitchell) [inline libsplinter/src/peer/builder.rs:89]: s/will  wait/will wait/

File: libsplinter/src/peer/builder.rs [hunk 4]
```diff
@@ -75,6 +84,33 @@ impl PeerManagerBuilder {
...
+    /// peers endpoints
```
Comment (jsmitchell) [inline libsplinter/src/peer/builder.rs:90]: s/peers/peer's/

File: libsplinter/src/peer/builder.rs [hunk 5]
```diff
@@ -75,6 +84,33 @@ impl PeerManagerBuilder {
...
+    pub fn with_retry_frequency(mut self, retry_frequency: u64) -> Self {
+        self.retry_frequency = Some(retry_frequency);
+        self
+    }
+
+    /// Set the max_retry_frequency to use with the resulting `PeerManager`.
+    ///
+    /// The maximum time (in seconds) the `PeerManager` will wait before retrying a pending
+    /// peers endpoints
```
Comment (jsmitchell) [inline libsplinter/src/peer/builder.rs:99]: s/peers/peer's/

File: libsplinter/src/peer/builder.rs (hunk 3 above)
Comment (agunde406) [reply libsplinter/src/peer/builder.rs]: fixed

File: libsplinter/src/peer/builder.rs (hunk 4 above)
Comment (agunde406) [reply libsplinter/src/peer/builder.rs]: fixed

File: libsplinter/src/peer/builder.rs (hunk 5 above)
Comment (agunde406) [reply libsplinter/src/peer/builder.rs]: fixed

**Pull request merged**